core/lib/tasks/solidus/split_promotions_with_any_match_policy.rake

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+namespace :solidus do
+  desc "Split Promotions with 'any' match policy"
+  task split_promotions_with_any_match_policy: :environment do
+    Spree::Promotion.where(match_policy: :any).includes(:promotion_rules).all.each do |promotion|
+      if promotion.promotion_rules.length <= 1
+        promotion.update!(match_policy: :all)
+      elsif promotion.active?
+        promotion.rules.map do |rule|
+          new_promotion = promotion.dup
+          new_promotion.promotion_rules = [rule]
+          new_promotion.match_policy = "all"
+          new_promotion.promotion_actions = promotion.actions.map do |action|
```

Comment:
It seems like this is going to create many extra actions 🤔 For example, having 3 rules will cause 12 actions to be created due to the nesting. Though, it will end up with the correct set of actions due to `promotion_actions =`